pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/dictionary/DictionaryIndexType.java

```diff
@@ -162,10 +167,11 @@ public ColumnConfigDeserializer<DictionaryIndexConfig> createDeserializer() {
       Set<String> varLength = new HashSet<>(
           ic.getVarLengthDictionaryColumns() == null ? Collections.emptyList() : ic.getVarLengthDictionaryColumns()
       );
+
+      // Intern configs can only be used through FieldConfigLists.
       Function<String, DictionaryIndexConfig> valueCalculator =
-          column -> new DictionaryIndexConfig(onHeap.contains(column), varLength.contains(column));
-      return Sets.union(onHeap, varLength).stream()
-          .collect(Collectors.toMap(Function.identity(), valueCalculator));
+          column -> new DictionaryIndexConfig(onHeap.contains(column), varLength.contains(column), Intern.DISABLED);
+      return Sets.union(onHeap, varLength).stream().collect(Collectors.toMap(Function.identity(), valueCalculator));
```

Comment:
Minor: Like most fluent DSLs, usually stream operators are written in separated lines. So I would prefer to keep the previous code style. Also, if we add the older constructor to DictionaryIndexConfig, we don't actually need to modify these lines (neither 117-119) beyond adding a comment saying that intern configs can only be used through FieldConfigLists, which I think it is a nice addition